sound/soc/sof/ipc4-topology.c

```diff
@@ -1837,10 +1836,19 @@ static int sof_ipc4_prepare_src_module(struct snd_sof_widget *swidget,
 	if (ret < 0)
 		return ret;
 
-	in_fmt = &available_fmt->input_pin_fmts[ret].audio_fmt;
```

Comment:
@plbossart I should have really said multiple rates. But if the DAI supports multiple rates, whats the need for an SRC in this case?

Reply:
The application may want to play a 44.1kHz tone on the BT link, and that link is either at 8 or 16 kHz depending on the BT profile.

It's a bit hypothetical but the DAI rate is not necessarily determined by the PCM rate, external bus or link limitations may have to be factored in.

Reply:
In this case, what rate would the SRC pick for it's output @plbossart? And what criteria would it apply to choose it? I think we're asking for a bit too much from the kernel driver in this case. Do we really want to complicate the audio format selection logic for this case? Won't we simply have 2 seaprate PCMs for the 2 different profiles?

Reply:
I don't necessarily have an answer here, but instead of making this strong assumption "For playback, the SRC will only ever have a single output format based on what the DAI supports." we can have a more reasonable "the SRC selection logic can only deal with a DAI with a single format' and throw errors in case the SRC is not connected to the DAI directly or the DAI has multiple formats.

Reply:
Could we take an approach with steps? This PR supports current testing and basic usage needs for SRC, before everything is broken with IPC4, and we could not see it in code reviews. We should next construct a test topology with multiple playback SRC output rates and do another fix round for kernel. It's hard to see all possible problems before we experience them for real. But please don't delay merge of this. Fundamental things get broken very quickly when we don't have tests.

Reply:
I've updated the comment now @plbossart . Let me know if this looks OK.